src/app/products/product-card-rating/product-card-rating.component.ts

```diff
@@ -0,0 +1,14 @@
+import { Component, Input } from '@angular/core';
+import { Product } from 'src/app/types';
+
+@Component({
+  selector: 'app-product-card-rating',
+  templateUrl: './product-card-rating.component.html',
+  styleUrls: ['./product-card-rating.component.scss']
+})
+export class ProductCardRatingComponent {
+  @Input () public product!: Product;
```

Comment:
do you need whole product here? I think you need only `productRating: number`, nothing more. Product rating component don't use it. It's best practice to transfer only the data component use, nothing more.